### DISCUSSION TITLE
What buftype if nui windows?

### DISCUSSION THREAD

**Discussion by RAprogramm** [MunifTanjim/nui.nvim#243] [Q&A] [answered]
[Screencast_00003.webm](https://user-images.githubusercontent.com/70325462/225836487-31d74148-0ede-421f-a9a0-17b617d4deb8.webm)

when opening any nui window - lualine looks like no good. to disable it i need to know filetype or buf type of nui. but i can't get it.
help me please

**Comment by RAprogramm** [comment]
`buf_options = { filetype = 'prompt' }` - this will decide my problem, BUT where i can paste it?
somewere in init options?

**Comment by MunifTanjim** [comment, marked as answer]
All the popup created by nui.nvim uses unlisted scratch buffer, created by `vim.api.nvim_create_buf(false, true)`.

If you want to set a specific `filetype`, you can pass `buf_options = { filetype = 'anything' }` to the popup_options table when creating the popup.

Also as a default Neovim behavior - for any popup `buftype` is `nofile`, and for input `buftype` is `prompt`.

**Reply by RAprogramm** [reply]
Thank you, but where I can put it? For example, neo-tree use nui_ui for filter files. So I should to put it globally in nui. How to do it? Something like override input and menu in vim with nui?

**Reply by MunifTanjim** [reply]
> So I should to put it globally in nui.

nui.nvim doesn't have any global option. This is just a library that other plugins use. If you want to configure something for `neo-tree` it's better to ask it on `neo-tree` repo.

> Something like override input and menu in vim with nui?

For `vim.ui.*` overrides, you can just put it inside the `popup_options` table. I see in your recording that you have set `win_options`, you can put `buf_options` just like that. e.g.

```lua
local popup_options = {
  -- ... other options
  buf_options = {
    -- ... buf options
  },
  win_options = {
    -- ... win options
  },
}

**Reply by RAprogramm** [reply]
Thank you for your help! Good luck to you
